### PR TITLE
Cust 1841 show help app to permissioned users

### PR DESCRIPTION
**Pull request by darushka**
This PR adds 2 new services:
BootstrapAccount (grabs all nessesary data for defining access to core apps and pages)
Access (determs what access is needed for core apps)

and updates Toolbar and Sidenav services to use $peachAccess.

### CODE REVIEW

File: dist/peach.js [hunk 1]
```diff
@@ -319,6 +320,52 @@ return /******/ (function(modules) { // webpackBootstrap
 	      return deferred.promise;
 	    }
 
+	    function getDocs(doc_id) {
+	      var deferred = $q.defer();
+	      var activeAccount = $peachSession.getActiveAccount();
+
+	      if (!activeAccount) {
+	        deferred.reject({ "error": { "name": "No Active Account", "message": "An active account is not found, so we are unable to look up day parts." } });
```
Comment (jsmylnycky) [inline dist/peach.js]: This should probably say unable to look up docs.

File: src/providers/account/account.js [hunk 2]
```diff
@@ -91,6 +92,53 @@ class PeachAccountProvider {
         return deferred.promise;
       }
 
+      function getDocs(doc_id) {
+        let deferred = $q.defer();
+        let activeAccount = $peachSession.getActiveAccount();
+
+        if(!activeAccount) {
+          deferred.reject({"error":{"name":"No Active Account", "message":"An active account is not found, so we are unable to look up day parts."}});
```
Comment (jsmylnycky) [inline src/providers/account/account.js]: This should probably say unable to look up docs.

**Comment by jsmylnycky** [comment]
:+1:  soon as you fix up that error message.

**Comment by againsley** [comment]
Nice work Darya. :+1:

File: src/services/access/access.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+        deferred.resolve(this.coreApps);
+      }, (error) => {
+        deferred.reject(false);
+      });
+
+      return deferred.promise;
+  }
+
+  getConfigAccess() {
```
Comment (DarrylD) [inline src/services/access/access.js]: Some commenting in this method would be nice, a lot of conditions to read through.

File: src/services/access/access.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        deferred.resolve(appAccess);
+      }, (error) => {
+        deferred.reject(false);
+      });
+
+      return deferred.promise;
+  }
+
+  getCoreApps() {
+    let deferred = this.$q.defer();
```
Comment (DarrylD) [inline src/services/access/access.js]: Do we need this deferred? `this.$peachBootstrapAccount.getData()` returns a promise.

File: src/services/access/access.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    this.docs = [];
+    this.hasConfig = false;
+    this.hasReports = false;
+    this.isAdmin = false;
+    this.user = null;
+    this.userPrefsByKey = {};
+  }
+
+  getAppAccess(key) {
+    let deferred = this.$q.defer();
```
Comment (DarrylD) [inline src/services/access/access.js]: Is this deferred needed? `this.$peachBootstrapAccount.getData()` returns a promise.

File: src/services/bootstrap-account/bootstrap-account.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        deferred.reject(error);
+      });
+
+    return deferred.promise;
+  }
+
+  loadCorePermissions() {
+    let deferred = this.$q.defer();
+
+    this.$peach.api('apps/'+this.coreApp.key+'/permissions').find(null, {
```
Comment (DarrylD) [inline src/services/bootstrap-account/bootstrap-account.js]: Can we put a line break on `.find` and `.then`? Just to keep consistent with semantics.

File: src/providers/account/account.js (hunk 2 above)
Comment (DarrylD) [inline src/providers/account/account.js]: Can we abstract that error object and make it a variable? Would be easier to read then have it inline.

File: src/directives/sidenav/sidenav.controller.js [hunk 7]
```diff
@@ -113,11 +114,28 @@ class SidenavController {
     let _this = this;
```
Comment (rynomars) [inline src/directives/sidenav/sidenav.controller.js]: @darushka Should we go ahead and remove _this while we are in this file? So we stop using let _this = this. and just use this.

File: src/directives/toolbar/toolbar.controller.js [hunk 8]
```diff
@@ -66,70 +65,61 @@ class ToolbarController {
     });
   }
 
-  activateShowConfig() {
+  activateCoreApps() {
     let _this = this;
```
Comment (rynomars) [inline src/directives/toolbar/toolbar.controller.js]: Same here. Can we eliminate let _this = this; and just use this.

File: src/providers/account/account.js [hunk 9]
```diff
@@ -91,6 +92,53 @@ class PeachAccountProvider {
         return deferred.promise;
       }
 
+      function getDocs(doc_id) {
```
Comment (rynomars) [inline src/providers/account/account.js]: @darushka should doc_id be camel case, docId?

**Comment by rynomars** [comment]
:thumbsup:

File: src/providers/account/account.js (hunk 9 above)
Comment (darushka) [inline src/providers/account/account.js]: @rynomars 
I coded it like in other methods of this file to keep consistant with the rest of the code. I guess the underscore case in this file makes sense as it matches the API property which we pass as paramerter when using those methods. But let ask @jsmylnycky opinion on this.

File: src/services/access/access.js (hunk 4 above)
Comment (darushka) [inline src/services/access/access.js]: Fixed. See commit https://github.com/peachworks/ngPeach/commit/11b43e6a0029af3f201517e6c53aff91530b2789

File: src/services/access/access.js (hunk 5 above)
Comment (darushka) [inline src/services/access/access.js]: Fixed. See commit https://github.com/peachworks/ngPeach/commit/11b43e6a0029af3f201517e6c53aff91530b2789

File: src/services/bootstrap-account/bootstrap-account.js (hunk 6 above)
Comment (darushka) [inline src/services/bootstrap-account/bootstrap-account.js]: Fixed. See commit https://github.com/peachworks/ngPeach/commit/11b43e6a0029af3f201517e6c53aff91530b2789

File: src/directives/sidenav/sidenav.controller.js (hunk 7 above)
Comment (darushka) [inline src/directives/sidenav/sidenav.controller.js]: Fixed. See commit https://github.com/peachworks/ngPeach/commit/11b43e6a0029af3f201517e6c53aff91530b2789

File: src/directives/toolbar/toolbar.controller.js (hunk 8 above)
Comment (darushka) [inline src/directives/toolbar/toolbar.controller.js]: Fixed. See commit https://github.com/peachworks/ngPeach/commit/11b43e6a0029af3f201517e6c53aff91530b2789

File: src/providers/account/account.js (hunk 9 above)
Comment (jsmylnycky) [inline src/providers/account/account.js]: @rynomars @darushka Darya is correct that we are using snake case for other parameters throughout the ngPeach providers.

Yes we are allowing them to remain snaked in the case where it is a value provided by the API, such as working with an object directly returned by api.find(), we may want to consider updating these providers to use proper camelcase.

I don't think this is the PR to do it in, however, so I'm fine with it remaining as is.

**Comment by DarrylD** [comment]
:+1:

**Comment by jsmylnycky** [comment]
:+1:  after the refactor

**Comment by darushka** [comment]
@jsmylnycky 
Please review my last commit as a solution to handle the API error

**Comment by jsmylnycky** [comment]
Released v0.5.0

**Pull request closed without merging**